mempool/src/feerate.rs

```diff
@@ -0,0 +1,54 @@
+use common::primitives::amount::Amount;
+
+lazy_static::lazy_static! {
+    pub(crate) static ref INCREMENTAL_RELAY_FEE_RATE: FeeRate = FeeRate::new(1000);
+    pub(crate) static ref INCREMENTAL_RELAY_THRESHOLD: FeeRate = FeeRate::new(500);
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) struct FeeRate {
+    tokens_per_kb: u128,
+}
+
+impl FeeRate {
+    pub(crate) fn new(tokens_per_kb: u128) -> Self {
+        Self { tokens_per_kb }
+    }
+
+    pub(crate) fn of_tx(fee: Amount, tx_size: usize) -> Self {
+        let tokens_per_byte = Self::div_up(fee.into(), tx_size);
+        Self {
+            tokens_per_kb: tokens_per_byte * 1000,
+        }
+    }
+
+    pub(crate) fn compute_fee(&self, size: usize) -> Amount {
+        Amount::from(self.tokens_per_kb * u128::try_from(size).unwrap() / 1000)
+    }
```

Comment:
I don't think this is guaranteed to be precise, is it? Does the following always hold?
```rust
assert_eq!(Feerate::of_tx(amt, size).compute_fee(size), amt)
```

Reply:
Yes, you're right, this does not always hold.